src/Conversion/ONNXToKrnl/Quantization/QuantizeLinear.cpp

```diff
@@ -71,15 +71,31 @@ void emitQuantizationLinearScalarParameters(ConversionPatternRewriter &rewriter,
   DimsExpr outputAF;
   outputAF.emplace_back(zero);
 
-  // faster than original loop on z16, takes 124us for 64k vals
-  // Allocate output buffers.
-  MemRefType flatBufferType = llvm::cast<MemRefType>(flatInput.getType());
-  Value flatBuffer = create.mem.alignedAlloc(flatBufferType, flatInputDims);
-  DimsExpr bufferAF;
-  bufferAF.emplace_back(zero);
-
+  Type inputElementType = inputType.getElementType();
+  unsigned inputWidth;
+  if (isa<Float32Type>(inputElementType))
+    inputWidth = 32;
+  else if (isa<Float64Type>(inputElementType))
+    inputWidth = 64;
+  else
+    llvm_unreachable("unsupported input type");
+  IntegerType quantizedIntType = cast<IntegerType>(quantizedElementType);
+  bool isSigned = quantizedIntType.isSignless() || quantizedIntType.isSigned();
+  Type quantizedElementTypeInputSized;
+  if (isSigned) {
+    // Cannot use getIntegerType(inputWidth, true) as it returns signed ints.
```

Comment:
If you would like to create `i32` or `i64`, I guess we can use `getIntegerType(inputWidth)` that has a single parameter. Meanwhile `getIntegerType(inputWidth, true/false` with two parameters will emit `si32`/`ui32`, `si64`/`ui64`.